src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -841,5 +848,25 @@ object PruneDeadFields {
       }
     }
   }
+
+  def upcastTable(ir: TableIR,
+    rType: TableType,
+    upcastRow: Boolean = true,
+    upcastGlobals: Boolean = true): TableIR = {
+    if (ir.typ == rType)
+      ir
+    else {
+      var table = ir
+      if (upcastRow && ir.typ.rowType != rType.rowType) {
+        table = TableMapRows(table, upcast(Ref("row", table.typ.rowType), rType.rowType), None, None)
```

Comment:
this unkeys the table

Reply:
doesn't it?

Reply:
I thought "none" left the key stuff unchanged?

Reply:
Unless you're talking about something else

Reply:
nope

Reply:
see https://github.com/hail-is/hail/issues/3881

Reply:
```
  val typ: TableType = {
    val newRowType = newRow.typ.asInstanceOf[TStruct]
    child.typ.copy(rowType = newRowType, key = newKey)
  }
```